bioptim/dynamics/configure_problem.py

```diff
@@ -312,6 +316,68 @@ def configure_contact_function(ocp, nlp, dyn_func: Callable, **extra_params):
             legend=all_contact_names,
         )
 
+    @staticmethod
+    def configure_soft_contact_function(ocp, nlp):
+        """
+        Configure the soft contact sphere
+
+        Parameters
+        ----------
+        ocp: OptimalControlProgram
+            A reference to the ocp
+        nlp: NonLinearProgram
+            A reference to the phase
+        """
+        nlp.soft_contact_forces_func = []
+        for i_sc in range(nlp.model.nbSoftContacts()):
+            soft_contact = nlp.model.softContact(i_sc)
+            n = int(nlp.states.mx_reduced.shape[0] / 2)
+            softcontact_force_func = (
+                biorbd.SoftContactSphere(soft_contact)
+                .computeForceAtOrigin(nlp.model, nlp.states.mx_reduced[:n], nlp.states.mx_reduced[n:])
+                .to_mx()
+            )
+
+            nlp.soft_contact_forces_func.append(
+                Function(
+                    "soft_contact_forces_func",
+                    [nlp.states.mx_reduced, nlp.controls.mx_reduced, nlp.parameters.mx],
+                    [softcontact_force_func],
+                    ["x", "u", "p"],
+                    ["soft_contact_forces"],
+                ).expand()
+            )
```

Comment:
Casadi function are heavy on everything (time, RAM, etc.). If there exist a function a gets all of these at once that you can interface, and then select the column in the for loop, it would allow to move the Function creation outside of the loop saving a loooooot of RAM while create SX, or when computing the jacobian

Reply:
Challenge accepted, but what if i need to do a graphs by contact sphere ??

Reply:
You can call the function by selecting columns